packages/ts/file-router/test/vite-plugin/vite-plugin.spec.ts

```diff
@@ -35,54 +96,147 @@ describe('@vaadin/hilla-file-router', () => {
+      const viewFileModule = createMockEnvironmentModuleNode(viewFilePath);
+      viewFileModule.importers.add(fileRoutesModule);
+      const modulesToUpdate = await hotUpdate({ type: 'update', file: viewFilePath, modules: [viewFileModule] });
+      expect(mockEnvironment.fetchModule).to.be.calledWith(fileRoutesTsPath);
+      // HMR for routes and view file together.
+      expect(modulesToUpdate).to.not.equal(undefined);
+      expect(modulesToUpdate!.length).to.equal(2);
+      expect(modulesToUpdate![0]).to.eq(fileRoutesModule);
+      expect(modulesToUpdate![1]).to.eq(viewFileModule);
+      expect(mockEnvironment.hot.send).to.not.be.called;
+    });
+
+    it('should send hot reload including file routes when a view file imported in file routes is added', async () => {
+      expect(mockEnvironment.hot.send).to.not.be.called;
+      const viewFilePath = fileURLToPath(new URL('file.tsx', viewsDir));
+      const viewFileModule = createMockEnvironmentModuleNode(viewFilePath);
+      viewFileModule.importers.add(fileRoutesModule);
+      const modulesToUpdate = await hotUpdate({ type: 'create', file: viewFilePath, modules: [viewFileModule] });
+      expect(mockEnvironment.fetchModule).to.be.calledWith(fileRoutesTsPath);
+      // HMR for routes and view file together.
+      expect(modulesToUpdate).to.not.equal(undefined);
+      expect(modulesToUpdate!.length).to.equal(2);
+      expect(modulesToUpdate![0]).to.eq(fileRoutesModule);
+      expect(modulesToUpdate![1]).to.eq(viewFileModule);
+      expect(mockEnvironment.hot.send).to.not.be.called;
+    });
+
+    it('should send hot reload for only file routes when a view file not imported anywhere changes', async () => {
+      expect(mockEnvironment.hot.send).to.not.be.called;
+      const viewFilePath = fileURLToPath(new URL('file.tsx', viewsDir));
+      const viewFileModule = createMockEnvironmentModuleNode(viewFilePath);
+      const modulesToUpdate = await hotUpdate({ type: 'update', file: viewFilePath, modules: [viewFileModule] });
+      expect(mockEnvironment.fetchModule).to.be.calledWith(fileRoutesTsPath);
+      // HMR for only file routes.
+      expect(modulesToUpdate).to.not.equal(undefined);
+      expect(modulesToUpdate!.length).to.equal(1);
+      expect(modulesToUpdate![0]).to.eq(fileRoutesModule);
+      expect(mockEnvironment.hot.send).to.not.be.called;
+    });
+
+    it('should send hot reload for only file routes when a view file not imported anywhere is removed', async () => {
+      expect(mockEnvironment.hot.send).to.not.be.called;
+      const viewFilePath = fileURLToPath(new URL('file.tsx', viewsDir));
+      const viewFileModule = createMockEnvironmentModuleNode(viewFilePath);
+      const modulesToUpdate = await hotUpdate({ type: 'delete', file: viewFilePath, modules: [viewFileModule] });
+      expect(mockEnvironment.fetchModule).to.be.calledWith(fileRoutesTsPath);
+      // HMR for only file routes.
+      expect(modulesToUpdate).to.not.equal(undefined);
```

Comment:
Here and in other places:
```suggestion
      expect(modulesToUpdate).to.not.be.undefined;
```